src/test/java/com/uid2/shared/secure/AttestationTokenTest.java

```diff
@@ -67,6 +67,20 @@ public void testAttestationTokenNew() {
         Assertions.assertTrue(reconstructToken.getExpiresAt().isAfter(expiryLowerBound));
         Assertions.assertTrue(reconstructToken.getExpiresAt().isBefore(expiryUpperBound));
     }
+    @Test
+    public void testAttestationTokenNewDefaultLifetime() {
+        final long defaultLifetime = 7200;
+        final Instant expiryLowerBound = Instant.now().plusSeconds(defaultLifetime + (5 * 60));
+        final Instant expiryUpperBound = Instant.now().plusSeconds(defaultLifetime + (10 * 60));
+        final AttestationTokenService ats = new AttestationTokenService(ENCRYPTION_KEY, SALT);
+
+        final String attestationToken = ats.createToken("userToken");
+        Assertions.assertTrue(ats.validateToken("userToken", attestationToken));
+
+        final AttestationToken reconstructToken = AttestationToken.fromEncrypted(attestationToken, ENCRYPTION_KEY, SALT);
+        Assertions.assertTrue(reconstructToken.getExpiresAt().isAfter(expiryLowerBound));
```

Comment:
We could make the tests more deterministic by constructor injecting a random number generator class and a clock into the AttestationTokenService instead of using statics. Then we could assert on exact equality of expected output rather than rely on bounds.

If you prefer to keep the bounds though, it's better to use AssertJ's `assertThat(reconstructToken.getExpiresAt()).isBefore(expiryUpperBound)` from `org.assertj.core.api.Assertions.assertThat` since this will give us a more useful error message when the test fails than `expected false  to be true`.

Reply:
For your latter comment, it's probably better to use `assertThat(reconstructToken.getExpiresAt()).isBetween(expiryLowerBound, expiryUpperBound)`

Reply:
I agree, and have update the service as well as the tests. Left the constructors in there so we don't have to refactor Core and Optout at the same time.